app/models/application_record.rb

```diff
@@ -6,6 +6,10 @@ def format_total_revenue
   end
 
   def format_unit_price
-    (self.unit_price.to_f / 100).to_fs(:currency)
+    if self.respond_to?(:total_item_revenue)
+      (self.total_item_revenue.to_f / 100).to_fs(:currency)
+    else
```

Comment:
Completely forgot that respond_to? was an option. Nice usage! Will have to keep that one in mind more in the future.